### PR TITLE
[IMP] l10n_uk* : rename uk to gb

### PR DESCRIPTION
**Pull request by jbw-odoo**
Rename "uk" instances to "gb" in order to respect norm ISO 3166-1 alpha-2 code

Task: 2373769

### CODE REVIEW

File: addons/l10n_cl/data/res.country.csv [hunk 1]
```diff
@@ -149,7 +149,7 @@ base.tv,419
 base.tz,135
 base.ua,559
 base.ug,136
-base.uk,510
+base.gb,510
```
Comment (william-andre) [inline addons/l10n_cl/data/res.country.csv:152]: Keep it sorted alphabetically by `id`

Comment (william-andre) [inline addons/l10n_cl/data/res.country.csv:152]: Keep it sorted alphabetically by `id`

File: addons/l10n_gb/data/account_chart_template_data.xml [hunk 2]
```diff
@@ -2,7 +2,7 @@
 <odoo>
     <data noupdate="1">
         <function model="account.chart.template" name="try_loading">
-            <value eval="[ref('l10n_uk.l10n_uk')]"/>
+            <value eval="[ref('l10n_gb.l10n_gb')]"/>
```
Comment (william-andre) [inline addons/l10n_gb/data/account_chart_template_data.xml:5]: Something more explicit for the chart template xml id would be nice, like `l10n_gb.l10n_gb_chart_template`. It seems like the right time to do it.

**Review by william-andre — COMMENTED** [review]
First, I'm not sure we should go further with this as we are not sure the accounting is different from Northern Ireland from the rest of the UK. Checking with TSB.
It seems the only difference between GB and NI are the taxes, so maybe creating 2 modules `l10n_ni` and `l10n_gb` that both inherit `l10n_uk` would make sense.

Runbot is red btw.

___________________________
For the internal part: @tbe-odoo 
https://github.com/odoo/internal/blob/6abd61a3058f245b652c3abc85b2280e48cd2180/private/openerp_enterprise/models/openerp_enterprise.py#L3551
Uses the country code `UK` to compute `is_eu`. The country code hasn't changed with this task as it was already `GB`. This is maybe to be considering Brexit too...
The controller routes of https://github.com/odoo/internal/blob/master/private/openerp_l10n_uk_hmrc_proxy/controllers/hmrc_proxy.py seem to be alright, but cleaning it might be better. I'll let you decide if you want the code to be clean or not.

____________________________
On another note, while reviewing this I also noticed our `sale_amazon` module is not up to date https://github.com/odoo/odoo/issues/64837

**Comment by jbw-odoo** [comment]
> First, I'm not sure we should go further with this as we are not sure the accounting is different from Northern Ireland from the rest of the UK. Checking with TSB.
> It seems the only difference between GB and NI are the taxes, so maybe creating 2 modules `l10n_ni` and `l10n_gb` that both inherit `l10n_uk` would make sense.

@tsb-odoo ?

**Comment by william-andre** [comment]
@jbw-odoo We have discussed it with TSB, and we decided to freeze the task while gathering more info; I added a not on the task last week

**Comment by tbe-odoo** [comment]
> First, I'm not sure we should go further with this as we are not sure the accounting is different from Northern Ireland from the rest of the UK. Checking with TSB.
> It seems the only difference between GB and NI are the taxes, so maybe creating 2 modules `l10n_ni` and `l10n_gb` that both inherit `l10n_uk` would make sense.
> 
> Runbot is red btw.
> 
> For the internal part: @tbe-odoo
> https://github.com/odoo/internal/blob/6abd61a3058f245b652c3abc85b2280e48cd2180/private/openerp_enterprise/models/openerp_enterprise.py#L3551
> Uses the country code `UK` to compute `is_eu`. The country code hasn't changed with this task as it was already `GB`. This is maybe to be considering Brexit too...
> The controller routes of https://github.com/odoo/internal/blob/master/private/openerp_l10n_uk_hmrc_proxy/controllers/hmrc_proxy.py seem to be alright, but cleaning it might be better. I'll let you decide if you want the code to be clean or not.
> 
> On another note, while reviewing this I also noticed our `sale_amazon` module is not up to date #64837

Hello @william-andre , the code you have mentioned on the internal repository isn't used anymore (dead code), so it shouldn't mess up with the brexit

**Comment by rdeodoo** [comment]
Task cancelled 7 months ago

**Pull request closed without merging**